weblate/screenshots/models.py

```diff
@@ -76,3 +86,29 @@ def change_screenshot_assignment(sender, instance, action, **kwargs):
         name="UnusedScreenshot"
     ).exists():
         component_alerts.delay([instance.pk])
+
+
+@receiver(vcs_post_update)
+def update_screenshots_from_repo(sender, component, previous_head: str, **kwargs):
+    screenshots = Screenshot.objects.filter(translation__component=component)
+    repo = component.repository
+
+    # Todo: Add suport for all supported VCSes
+    changed_files = repo.list_changed_files("refspec")
+
+    for s_shot in screenshots:
+        f_name = s_shot.repository_filename
+        if f_name in changed_files and component.changed_screenshot_files(f_name):
```

Comment:
```suggestion
        if f_name in changed_files:
```

I don't see need for this. Any matching screenshots should be updated. Component.screenshot_filemask should be used for adding new screenshots.

PS: This condition should not be needed at all, see above comment.